app/components/SearchableMetricSelector.js

```diff
@@ -1,25 +1,62 @@
 import { useEffect, useState } from 'react'
 
+function SortDictionary(dictionary){
+    var sortedTagsArr = Object.entries(dictionary).sort((a,b) => b[1].length - a[1].length)
+    var retArr = [];
+    for(let i = 0; i < sortedTagsArr.length; i++){
+      var tempArr = sortedTagsArr[i]
+      var tempDict = {};
+      tempDict[tempArr[0]] = tempArr[1]
+      retArr.push(tempDict)
+    }
+    return retArr
+  }
+  
+function CreateTagDictionary(supportedMetrics){
+  const groups = [...new Set(Object.keys(supportedMetrics).map(q => supportedMetrics[q].dataset))];
```

Comment:
I think what you are trying to do here is get a list of the unique group IDs. Write that in a comment above to help make this line more readable.